django_enum_choices/fields.py

```diff
@@ -52,8 +52,14 @@ def __init__(self, enum_class: Enum, choice_builder=value_value, **kwargs):
             validator for validator in self.validators
             if not isinstance(validator, MaxLengthValidator)
         ]
+
+        should_validate_attribute_name = self.should_validate_attribute_name(**kwargs)
+
         self.validators.append(
-            EnumValueMaxLengthValidator(kwargs['max_length'])
+            EnumValueMaxLengthValidator(
+                limit_value=kwargs['max_length'],
+                validate_attribute_name=should_validate_attribute_name
```

Comment:
I think we can pass `self.get_prep_value` here instead. That should handle custom choice builders as well.